portfolio/src/main/java/com/google/sps/servlets/CommentsDatastore.java

```diff
@@ -0,0 +1,70 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+public class CommentsDatastore {
+
+  public class Comment {
+    private String comment;
+    private long timestamp;
+
+    public Comment(String comment, long timestamp) {
+      this.comment = comment;
+      this.timestamp = timestamp;
+    }
+  } 
+
+    public CommentsDatastore() {}
+    
+    public List<Comment> fetchComments(int maxNumOfComments) {
+      Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      List<Entity> results = datastore.prepare(query)
+        .asList(FetchOptions.Builder.withLimit(maxNumOfComments));
+      
+      List<Comment> comments = new ArrayList<>();
+      for (Entity entity : results) {
+        String comment = (String) entity.getProperty("comment");
+        long timestamp = (long) entity.getProperty("timestamp");
+        Comment newComment = new Comment(comment, timestamp);
+        comments.add(newComment);
+      }
+
+      return comments;
+    }
+
+    public Comment addComment(String comment, int maxNumOfComments) {
+      long timestamp = System.currentTimeMillis();
+
+      if (comment != null && !comment.isEmpty()) {
+        Entity commentEntity = new Entity("Comment");
+        commentEntity.setProperty("comment", comment);
+        commentEntity.setProperty("timestamp", timestamp);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(commentEntity);
+      }
+
+      Comment newComment = new Comment(comment, timestamp);
```

Comment:
see comment above about argument check. you don't want to create a comment if your inputs are invalid. so you want to throw / return early if your inputs are incorrect ("guard" clause):
https://en.wikipedia.org/wiki/Guard_(computer_science)#:~:text=Regardless%20of%20which%20programming%20language,which%20avoids%20null%2Dpointer%20failures.